### ISSUE TITLE
IOS No Subscription Confirm Popup on IAP Version > 9.0.0

### ISSUE THREAD

**Issue by aesculus** [j3k0/cordova-plugin-purchase#1279]
### system info

Please report your OS, OS version, Compiling System and Version, used Plugin Version and any helpful information that can help us to resolve the problem.

In example:
```
OSX Catalina
XCODE 12.4 but Deployment Target is 10.0
Cordova 10.0, iOS 5.1.1
Device: iPhone iOS 14.8.1
Plugin Version: Anything greater than 9.0.0
```
### Expected behavior
The subscription Confirm Popup will display after 3 seconds when making a subscription purchase from the first popup.
### Observed behavior
No popup after waiting more than 10 seconds.
### Steps to reproduce
Request a subscription purchase, get dialog, order and then wait 10 seconds or more for confirmation popup. Nothing shows.

The attached image of popup does not appear.  Do I have some levels wrong? This app has been working fine for 2 years on 9.0.0 and on earlier versions without changing any of the code. Seems to work fine on 10.3 for Android.
![purchase_confirm](https://user-images.githubusercontent.com/283364/149452274-ae2e132e-eab5-424c-86b8-ad29402edfa8.png)

**Comment by mansourcodes** [comment]
Same issue, 

I create new Tester on Apple Developer, and use it on the Sandbox environment. that solve it.

But, you will need to create new tester every time the subscription reach the end.

**Comment by j3k0** [comment]
Can you share code and logs (with `store.verbosity = store.DEBUG` ? I'll take a look.

**Comment by aesculus** [comment]
Oh. Just saw this reply. I will try to get to it this week.

**Comment by aesculus** [comment]
Well I loaded up 10.3 and tried and it still fails when I select Purchase after my popup prompt. But unfortunately the log file shows no action.

Here is the log file during app startup if this helps at all:

     Date/Time: 09:40:53; Log Type: log; Data: [store.js] DEBUG: ios -> product dag_tesla_to_annual registered
     Date/Time: 09:40:53; Log Type: log; Data: [store.js] DEBUG: store.trigger -> triggering action refreshed
     Date/Time: 09:40:53; Log Type: log; Data: [store.js] DEBUG: ios -> initializing storekit
     Date/Time: 09:40:53; Log Type: log; Data: InAppPurchase[js]: setup ok
     Date/Time: 09:40:53; Log Type: log; Data: [store.js] INFO: ios -> storekit ready
     Date/Time: 09:40:53; Log Type: log; Data: [store.js] DEBUG: ios -> loading products
     Date/Time: 09:40:53; Log Type: log; Data: InAppPurchase[js]: load ["dag_tesla_to_yearly","dag_tesla_to_monthly","dag_tesla_to_annual"]
     Date/Time: 09:40:53; Log Type: log; Data: [store.js] DEBUG: state: dag_tesla_to_yearly -> registered
     Date/Time: 09:40:53; Log Type: log; Data: [store.js] DEBUG: ios -> product dag_tesla_to_yearly registered
     Date/Time: 09:40:53; Log Type: log; Data: [store.js] DEBUG: state: dag_tesla_to_monthly -> registered
     Date/Time: 09:40:53; Log Type: log; Data: [store.js] DEBUG: ios -> product dag_tesla_to_monthly registered
     Date/Time: 09:40:53; Log Type: log; Data: [store.js] DEBUG: state: dag_tesla_to_annual -> registered
     Date/Time: 09:40:55; Log Type: log; Data: InAppPurchase[js]: load ok: { valid:[{"id":"dag_tesla_to_annual","description":"A annual subscription fee for trip planning","introPrice":"$0.00","introPricePaymentMode":"FreeTrial","billingPeriodUnit":"Year","countryCode":"US","introPricePeriodUnit":"Day","discounts":[],"title":"Annual Subscription Fee","price":"$9.99","billingPeriod":1,"group":"20342256","priceMicros":9990000,"currency":"USD","introPricePeriod":7,"introPriceMicros":0},{"id":"dag_tesla_to_monthly","description":"A monthly subscription fee for trip planning","introPrice":"$0.00","introPricePaymentMode":"FreeTrial","billingPeriodUnit":"Month","countryCode":"US","introPricePeriodUnit":"Day","discounts":[],"title":"Monthly Subscription Fee","price":"$0.99","billingPeriod":1,"group":"20342256","priceMicros":990000,"currency":"USD","introPricePeriod":7,"introPriceMicros":0},{"id":"dag_tesla_to_yearly","description":"A Yearly Subscription Fee allows you to store and manage multiple trips created with EVTO. You may cancel your subscription within the first 30 days for no charge. This subscription will auto-renew before the subscription expires unless cancelled.","introPrice":"$0.00","introPricePaymentMode":"FreeTrial","billingPeriodUnit":"Year","countryCode":"US","introPricePeriodUnit":"Month","discounts":[],"title":"Yearly Subscription Fee","price":"$9.99","billingPeriod":1,"group":"20342256","priceMicros":9990000,"currency":"USD","introPricePeriod":1,"introPriceMicros":0}] invalid:[] }
     Date/Time: 09:40:55; Log Type: log; Data: [store.js] DEBUG: ios -> products loaded
     Date/Time: 09:40:55; Log Type: log; Data: [store.js] DEBUG: ios -> product dag_tesla_to_annual is valid (subscription_a)
     Date/Time: 09:40:55; Log Type: log; Data: [store.js] DEBUG: ios -> owned? false
     Date/Time: 09:40:55; Log Type: log; Data: [store.js] DEBUG: state: dag_tesla_to_annual -> valid
     Date/Time: 09:40:55; Log Type: log; Data: [store.js] DEBUG: ios -> product dag_tesla_to_monthly is valid (subscription_m)
     Date/Time: 09:40:55; Log Type: log; Data: [store.js] DEBUG: ios -> owned? false
     Date/Time: 09:40:55; Log Type: log; Data: [store.js] DEBUG: state: dag_tesla_to_monthly -> valid
     Date/Time: 09:40:55; Log Type: log; Data: [store.js] DEBUG: ios -> product dag_tesla_to_yearly is valid (subscription)
     Date/Time: 09:40:55; Log Type: log; Data: [store.js] DEBUG: ios -> owned? false
     Date/Time: 09:40:55; Log Type: log; Data: [store.js] DEBUG: state: dag_tesla_to_yearly -> valid
     Date/Time: 09:40:55; Log Type: log; Data: [store.js] DEBUG: update()
     Date/Time: 09:40:55; Log Type: log; Data: InAppPurchase[js]: loading appStoreReceipt
     Date/Time: 09:40:55; Log Type: log; Data: InAppPurchase[js]: infoPlist: com.xxxxx.teslato,6.6,117473280,????
     Date/Time: 09:40:55; Log Type: log; Data: [store.js] DEBUG: state: com.xxxxx.teslato ->
     Date/Time: 09:40:55; Log Type: log; Data: [store.js] DEBUG: state: com.xxxxx.teslato -> registered
     Date/Time: 09:40:55; Log Type: log; Data: [store.js] DEBUG: ios -> product com.xxxxx.teslato registered
     Date/Time: 09:40:55; Log Type: log; Data: [store.js] DEBUG: state: com.xxxxx.teslato -> approved
     Date/Time: 09:40:55; Log Type: log; Data: [store.js] DEBUG: state: com.xxxxx.teslato -> approved
     Date/Time: 09:40:56; Log Type: log; Data: InAppPurchase[js]: loading appStoreReceipt
     Date/Time: 09:40:56; Log Type: log; Data: InAppPurchase[js]: infoPlist: com.xxxxx.teslato,6.6,117473280,????
     Date/Time: 09:40:56; Log Type: log; Data: [store.js] DEBUG: state: com.xxxxx.teslato -> approved
     Date/Time: 09:40:56; Log Type: log; Data: [store.js] DEBUG: scheduleValidation()
     Date/Time: 09:40:58; Log Type: log; Data: [store.js] DEBUG: runValidation()
     Date/Time: 09:40:58; Log Type: log; Data: [store.js] DEBUG: ajax -> send request to https://www.xxxxx.com/services/check_purchase_b.php
     Date/Time: 09:40:59; Log Type: log; Data: [store.js] DEBUG: validator success, response: {"ok":false,"system":"sandbox","data":{"code":6778003}}
     Date/Time: 09:40:59; Log Type: log; Data: [store.js] DEBUG: verify -> {"success":false,"data":{"code":6778003}}
     Date/Time: 09:40:59; Log Type: log; Data: [store.js] DEBUG: verify -> error: {"code":6778003}
     Date/Time: 09:40:59; Log Type: log; Data: [store.js] DEBUG: ios -> product com.xxxxx.teslato expired
     Date/Time: 09:40:59; Log Type: log; Data: [store.js] DEBUG: ios -> product com.xxxxx.teslato owned=false
     Date/Time: 09:40:59; Log Type: log; Data: [store.js] DEBUG: state: com.xxxxx.teslato -> valid
     Date/Time: 09:40:59; Log Type: log; Data: [store.js] DEBUG: state: com.xxxxx.teslato -> valid
     Date/Time: 09:40:59; Log Type: log; Data: [store.js] DEBUG: store.trigger -> triggering action refresh-finished

**Comment by wsamoht** [comment]
I seem to be having the same problem testing with TestFlight. The app hasn't been released to the App Store yet. I am on an iPad running iOS 15.5. The IAP plugin version is 11.0.0.

Once a subscription expires, my validation server returns expired and the plugin does in fact call the `PURCHASE_EXPIRED` logic in `project.js` as I get the "Transaction expired:" message. I have also verified that the product's state gets set back to `valid`.

I then try to re-order the product but when calling `store.order` the native purchase dialog/confirmation never opens.

At the moment, the only way to re-subscribe is by going into the App Store and managing the subscription under the Sandbox account. I haven't had any issues on Android. I am not sure where to go from here.

**Comment by asaf-interator** [comment]
I also seem to have the same issue, after registering the products, when store.ready() === true and product.canPurchase === true and product.state === "valid", store.order(productId) only get updated to state == "requested" then stuck, the native purchase dialog/confirmation never opens.

It doesn't happen every time and it only happens on iOS.

**Comment by mattjamesvogt** [comment]
I am having the same issue, and it was working up until recently. Could it be something with apple?

**Comment by aesculus** [comment]
Any updates to this issue? I am still stuck on V9 for iOS users.

**Comment by wsamoht** [comment]
I gave up and switched to [RevenueCat](https://www.revenuecat.com). Since subscriptions are quite important for our app to make money, I rather pay a little to have support and a whole team behind the plugin. The integration with RevenueCat was much cleaner in my opinion as well.

I was quite disappointed at first as I had all the code done within my app and in the backend to handle webhooks from Google and Apple. But, in the long-term, I think we made the right decision.

**Comment by j3k0** [comment]
@aesculus

1. Can you share logs of what happens when you click call `order()`.
2. I see `store.update()` is called during startup, is it? That wouldn't be correct.

**Comment by aesculus** [comment]
I am  going to load up V11 and try it again as soon is Apple is done updating my machine. Been at it for 3 days now. :-(

I am not sure where that Date/Time: 09:40:55; Log Type: log; Data: [store.js] DEBUG: update() came from. Its not in my code.

**Comment by aussieEd** [comment]
I am now having the same issue so would value any tips. It's rather strange as the issue has started happening on a published app, where the in-app purchases were working well for several weeks and then the issue suddenly started.

**Comment by aesculus** [comment]
I finally got around to updating my app to the version 13.x of the plugin. Still had this problem. But noted (as above) I had a bunch of older receipts still in a non verified state. So I forced their verification on my back ended server. They were really old and maybe a mixture of Sandbox and regular receipts so it was all garbage.

I then deleted the app from the device and restarted it. The dialog then appeared again. Signed on to a Testflight account and then reordered the subscription and all was well again.
